src/MSBuildLocator/MSBuildLocator.cs

```diff
@@ -324,6 +330,44 @@ private static IEnumerable<VisualStudioInstance> GetInstances(VisualStudioInstan
         }
 
 #if NET46
+        /// <summary>
+        /// Gets CoreXT instances.
+        /// </summary>
+        /// <param name="visualStudioInstances">Optional <see cref="IList{VisualStudioInstance} "/> containing instances.  If none are specified, they are discovered through the Visual Studio Setup API.
+        /// This is only here for unit tests</param>
+        /// <param name="directoryExists">An optional <see cref="Func{String,Boolean} "/> to use when checking directory existence.  Should only be set by unit tests.</param>
+        /// <returns>A <see cref="VisualStudioInstance" /> object representing the Visual Studio instance if one is found, otherwise null.</returns>
+        internal static VisualStudioInstance GetCoreXTInstance(IList<VisualStudioInstance> visualStudioInstances = null, Func<string, bool> directoryExists = null)
+        {
+            var toolset = Environment.GetEnvironmentVariable("MsBuildToolset")?.Trim();
```

Comment:
Should we also have a check for a CoreXT environment variable like maybe `CoreXtExeVersion` to limit false positives?